vernac/declare.mli

```diff
@@ -119,9 +119,10 @@ val close_proof : opaque:opacity_flag -> keep_body_ucst_separate:bool -> Proof.t
 (** XXX: This is an internal, low-level API and could become scheduled
    for removal from the public API, use higher-level declare APIs
    instead *)
-type variable_declaration =
-  | SectionLocalDef of Evd.side_effects proof_entry
-  | SectionLocalAssum of { typ:types; impl:Glob_term.binding_kind; }
+module Variable_declaration : sig
+  type t
+  val make : typ:types -> impl:Glob_term.binding_kind -> t
```

Comment:
Shouldn't there be an optional body? Or maybe make this internal and have declare_variable take directly typ/impl

Reply:
It seems to me that the case using the entry is internal; indeed we could just modify `declare_variable` ? I'm OK with that if you feel it is cleaner.